### ISSUE TITLE
Bug in Algorithm Select

### ISSUE THREAD

**Issue by cliedeman** [containous/traefik#3821]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?
Upgraded to traefik 16.6.5 from 17.0.rc3 

### What did you expect to see?

ACME certificates renewing as expected.

### What did you see instead?
```
time="2018-08-24T13:10:46Z" level=error msg="Unable to obtain ACME certificate for domains \"foo.com\" detected thanks to rule \"Host:foo.com\" : unable to generate a certificate for the domains [foo.com]: acme: Error -&gt; One or more domains had a problem:\n[foo.com] invalid KeyType: \n"
```

### Output of `traefik version`: (_What version of Traefik are you using?_)
```
17.0.rc3
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?
kubernetes helm chart with acme.json config. Single instance
ACME enabled, challenge type: http-01

The issue is related to #3319. In my acme.json my KeyType string was empty (as expected). This should default correctly due [to](https://github.com/containous/traefik/blob/v1.7.0-rc3/provider/acme/account.go#L77) but its somehow getting through to lego causing this [log](https://github.com/xenolf/lego/blob/a2543a2fde2fafa4bc663e0058e59915967e8034/acme/crypto.go#L213)

I *think* this [line](https://github.com/containous/traefik/blob/dad0e75121bd3c19890ef399802daf858deda930/acme/acme.go#L430) should probabably do a check for an empty key and probably panic with a message to the user that the Key Type is missing. I am not sure if it is safe to default to 4096.

As I workaround I manually set the key type to 4096 inside my acme.json.

**Comment by ldez** [comment]
duplicate of https://github.com/containous/traefik/issues/3468#issuecomment-410192111

Already fixed by #3733

**Comment by cliedeman** [comment]
Thanks
